packages/components/src/navigation/stories/index.js

```diff
@@ -132,3 +136,73 @@ function Example() {
 export const _default = () => {
 	return <Example />;
 };
+
+function ComponentsCompositionExample() {
+	const [ activeItem, setActiveItem ] = useState( 'item-1' );
+	return (
+		<Navigation2>
+			{ ( { activeLevel, navigateTo, NavigationBackButton } ) => (
+				<>
+					<Navigation2Level
+						slug="root"
+						title="Home"
+						activeLevel={ activeLevel }
+					>
+						<Navigation2Item
+							slug="item-1"
+							title="Item 1"
+							onClick={ () => {
+								navigateTo( 'sub-menu' );
+							} }
+							activeItem={ activeItem }
+						/>
+					</Navigation2Level>
+
+					<Navigation2Level
+						slug="sub-menu"
+						title="Sub Menu"
+						activeLevel={ activeLevel }
+						navigateBack={
+							<NavigationBackButton
+								parentLevel="root"
+								label="Back to Home"
+							/>
+						}
+					>
+						<Navigation2Item
+							slug="item-2"
+							title="Item 2"
+							onClick={ () => {
+								navigateTo( 'nested-sub-menu' );
+							} }
+							activeItem={ activeItem }
+						/>
+					</Navigation2Level>
+
+					<Navigation2Level
+						slug="nested-sub-menu"
+						title="SubSub Menu"
+						activeLevel={ activeLevel }
+						navigateBack={
+							<NavigationBackButton
+								parentLevel="sub-menu"
+								label="Back to Sub Menu"
+							/>
```

Comment:
>and the implementation of the back button looks nasty.

Perhaps you're not finished with this just yet, but what advantage do you see by having each level designate its own back button?

Reply:
Alternatively we can get rid of exposing `NavigationBackButton` altogether in favor of just having a prop `parentLevel="root"`. `Navigation2Level` would be responsible for rendering the back button.